src/expr/src/scalar/func.rs

```diff
@@ -4732,11 +4735,18 @@ impl fmt::Display for BinaryFunc {
     // 'A' < 'AA' but 'AZ' > 'AAZ'.)
     is_monotone = (false, true),
 )]
-fn text_concat_binary<'a>(a: &str, b: &str, temp_storage: &'a RowArena) -> &'a str {
+fn text_concat_binary<'a>(
+    a: &str,
+    b: &str,
+    temp_storage: &'a RowArena,
+) -> Result<&'a str, EvalError> {
+    if a.len() + b.len() > MAX_STRING_FUNC_RESULT_BYTES {
+        return Err(EvalError::LengthTooLarge);
+    }
     let mut buf = String::with_capacity(a.len() + b.len());
     buf.push_str(a);
     buf.push_str(b);
-    temp_storage.push_string(buf)
+    Ok(temp_storage.push_string(buf))
 }
```

Comment:
You could simplify the function, too:
```suggestion
fn text_concat_binary<'a>(
    a: &str,
    b: &str,
) -> Result<String, EvalError> {
    if a.len() + b.len() > MAX_STRING_FUNC_RESULT_BYTES {
        return Err(EvalError::LengthTooLarge);
    }
    let mut buf = String::with_capacity(a.len() + b.len());
    buf.push_str(a);
    buf.push_str(b);
    buf
}
```